iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/digitaltwin/DigitalTwinClientTests.java

```diff
@@ -183,6 +190,84 @@ public void getDigitalTwin() {
         assertEquals(responseWithHeaders.body().getMetadata().getModelId(), E2ETestConstants.THERMOSTAT_MODEL_ID);
     }
 
+    // Open a multiplexed connection with two devices, each with a different model Id. Verify that their reported twin has
+    // the expected model Ids.
+    @Test
+    @StandardTierHubOnlyTest
+    public void getMultiplexedDigitalTwinsRegisteredBeforeOpen() throws IotHubException, IOException, URISyntaxException, MultiplexingClientException, InterruptedException {
+        if (protocol == MQTT || protocol == MQTT_WS) {
+            return; // multiplexing isn't supported over MQTT, so it can't be tested
+        }
+
+        MultiplexingClient multiplexingClient = new MultiplexingClient(deviceClient.getConfig().getIotHubHostname(), protocol);
+        List<DeviceClient> multiplexedDeviceClients = new ArrayList<>();
+        multiplexedDeviceClients.add(createDeviceClient(protocol, E2ETestConstants.THERMOSTAT_MODEL_ID));
+        multiplexedDeviceClients.add(createDeviceClient(protocol, E2ETestConstants.TEMPERATURE_CONTROLLER_MODEL_ID));
+
+        multiplexingClient.registerDeviceClients(multiplexedDeviceClients);
+
+        multiplexingClient.open();
+
+        // act
+        String thermostatDeviceId = multiplexedDeviceClients.get(0).getConfig().getDeviceId();
+        BasicDigitalTwin response = digitalTwinClient.getDigitalTwin(thermostatDeviceId, BasicDigitalTwin.class);
+        ServiceResponseWithHeaders<BasicDigitalTwin, DigitalTwinGetHeaders> responseWithHeaders =
+            digitalTwinClient.getDigitalTwinWithResponse(thermostatDeviceId, BasicDigitalTwin.class);
+
+        // assert
+        assertEquals(response.getMetadata().getModelId(), E2ETestConstants.THERMOSTAT_MODEL_ID);
+        assertEquals(responseWithHeaders.body().getMetadata().getModelId(), E2ETestConstants.THERMOSTAT_MODEL_ID);
+
+        // act
```

Comment:
I don't love unit tests that get overly complicated beyond a single pass at arrange/act/assert.

Should this be 2 unit tests? Can the acts and asserts be grouped together?

Reply:
This is an integration test. I can remove one of the "act" blocks here, though